arkouda/io.py

```diff
@@ -1617,3 +1643,94 @@ def read(
         )
     else:
         raise RuntimeError(f"Invalid File Type detected, {ftype}")
+
+
+def read_tagged_data(
+    filenames: Union[str, List[str]],
+    datasets: Optional[Union[str, List[str]]] = None,
+    iterative: bool = False,
+    strictTypes: bool = True,
+    allow_errors: bool = False,
+    calc_string_offsets=False,
+):
+    """
+    Read datasets from files and tag each record to the file it was read from.
+    File Type is determined automatically.
+
+    Parameters
+    ----------
+    filenames : list or str
+        Either a list of filenames or shell expression
+    datasets : list or str or None
+        (List of) name(s) of dataset(s) to read (default: all available)
+    iterative : bool
+        Iterative (True) or Single (False) function call(s) to server
```

Comment:
Should we say `"Cannot tag data with iterative read."` and also add a `RuntimeError` like in `read_hdf5/parquet`? Or maybe it would make more sense not to let them pass in a bool for `iterative` for the `read_tagged_data` function?

Reply:
Agreed. I removed the iterative param from the `read_tagged_data` function and hard-coded the calls to false."